File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_dict = {}
        for i, n in enumerate(nums):
            my_dict[n] = i;
        
        for i, n in enumerate(nums):
            diff = target - nums[i]

            if diff in my_dict.keys() and my_dict[diff] != i:
                return sorted([i, my_dict[diff]])

File: Data Structures & Algorithms/binary-search/submission-4.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, n = nums.length - 1;

        while(l <= n) {
            int m = l + ((n - l)/2);
            if(nums[m] > target) {
                n = m - 1;
            } else if (nums[m] < target) {
                l = m + 1;
            } else {
                return m;
            }
        }
        return -1;
    }
}
